modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/IgniteBaselineAffinityTopologyActivationTest.java

```diff
@@ -380,6 +381,99 @@ public void testNodeWithOldBltIsAllowedToJoinCluster() throws Exception {
         verifyBaselineTopologyOnNodes(verifier1, new Ignite[] {nodeA, nodeB, nodeC});
     }
 
+    /**
+     * Verifies that if the node is added to baseline to cluster in {@link ClusterState#ACTIVE_READ_ONLY} state, the cluster
+     * remains in this state.
+     *
+     * @throws Exception if failed.
+     */
+    @Test
+    public void testNodeJoinsToActiveReadOnlyCluster() throws Exception {
+        startGridWithConsistentId("A");
+        startGridWithConsistentId("B");
+
+        IgniteEx grid = grid("A");
+
+        grid.cluster().state(ClusterState.ACTIVE_READ_ONLY);
+
+        startGridWithConsistentId("C");
+
+        long topologyVersion = grid.cluster().topologyVersion();
+
+        try {
+            GridTestUtils.runAsync(
+                    () -> grid.cluster().setBaselineTopology(topologyVersion)
```

Comment:
Perhaps, it makes sense to check `void setBaselineTopology(Collection<? extends BaselineNode> baselineTop)` also.